### DISCUSSION TITLE
EACCESS Permission Denied Error

### DISCUSSION THREAD

**Discussion by ansonphong** [requarks/wiki#3323] [Error / Bug Report]
**Describe the bug**
When saving any page, I get this warning, see below. The saving actually happens, though it gives a warning then the next time I save it causes a merge conflict. Any help is appreciated.

**To Reproduce**
1. Import Wiki JS V1 pages
2. Edit a page
3. Press SAVE

**Expected behavior**
No error.

**Screenshots**
![image](https://user-images.githubusercontent.com/4825055/107500189-bdbd8c00-6b4a-11eb-9fc2-ec0ba3dcb409.png)

**Host Info (please complete the following information):**
 - OS: Ubuntu 18.04, Docker
 - Wiki.js version: 2.5.159
 - Database engine: MySQL

**Comment by scienceasdf** [comment]
Execute command `ls -l /var`, you will find if you have access to this directory. If this folder is not accessible, you can use `chown` or `chmod` to modify the access.

**Comment by ansonphong** [comment]
> Execute command `ls -l /var`, you will find if you have access to this directory. If this folder is not accessible, you can use `chown` or `chmod` to modify the access.

@scienceasdf Thanks, yes the directory is available. I even set `/var/wiki_vs_import` recursively to `777` with chmod and it still throws the same error every time I try to save any page. Any other suggestions?

**Comment by ansonphong** [comment]
I also found another error which is recurring which seems like it's related. When trying to rename any file in the system I get this error: `ENOENT: no such file or directory, stat '/var/wiki_vs_import/repo/[filename]'`

![image](https://user-images.githubusercontent.com/4825055/107647904-84f7e280-6c30-11eb-8698-1b659d6c68c0.png)


Again, that whole `/var/wiki_vs_import` folder is recursively set with permissions to `777` (temporarily to test) so that anybody can read/write.

**Comment by ansonphong** [comment]
This issue is also affecting renaming and other file modification operations within wiki.js. Any more pointers would be appreciated, why do you think it has anything to do with the wiki_vs_import folder??

**Comment by ansonphong** [comment]
The only way I was able to fix this was by running `chmod -R -f 755 /var` which exposes the server to potential security threats. Is there a map of what the specific folders permissions should be set to?
